### PR TITLE
[DOCS] Polish introductory section of introduction to smart contracts

### PR DESCRIPTION
**Pull request by ChrisChinchilla**
### Description

While recording a video to accompany this section I felt that the whole "introduction to smart contracts" page needed some polish, this is the first PR for the first section.

### Checklist
- [x] Code compiles correctly
- [x] All tests are passing
- [x] New tests have been created which fail without the change (if possible)
- [x] README / documentation was extended, if necessary
- [x] Changelog entry (if change is visible to the user)
- [x] Used meaningful commit messages

### CODE REVIEW

File: docs/introduction-to-smart-contracts.rst [hunk 1]
```diff
@@ -31,20 +31,19 @@ Storage Example
         }
     }
 
-The first line simply tells that the source code is written for
-Solidity version 0.4.0 or anything newer that does not break functionality
-(up to, but not including, version 0.7.0). This is to ensure that the
-contract is not compilable with a new (breaking) compiler version, where it could behave differently.
+The first line tells you that the source code is written for
+Solidity version 0.4.0, or newer that does not break functionality, up to, but not including, version 0.7.0.
```
Comment (chriseth) [inline docs/introduction-to-smart-contracts.rst]:
```suggestion
Solidity version 0.4.0, or a newer version of the language that does not break functionality, up to, but not including, version 0.7.0.
```

File: docs/introduction-to-smart-contracts.rst [hunk 2]
```diff
@@ -31,20 +31,19 @@ Storage Example
         }
     }
 
-The first line simply tells that the source code is written for
-Solidity version 0.4.0 or anything newer that does not break functionality
-(up to, but not including, version 0.7.0). This is to ensure that the
-contract is not compilable with a new (breaking) compiler version, where it could behave differently.
+The first line tells you that the source code is written for
+Solidity version 0.4.0, or a newer version of the language that does not break functionality, up to, but not including, version 0.7.0.
```
Comment (chriseth) [inline docs/introduction-to-smart-contracts.rst]: Actually, re-reading it - this is not even correct. It includes 0.5.0 which does break functionality. So I think it should be

`Solidity version 0.4.0 or a newer version of the language up to, but not including version 0.7.0`

Comment (ChrisChinchilla) [reply docs/introduction-to-smart-contracts.rst]: Updated

**Comment by chriseth** [comment]
Please rebase properly.

**Comment by ChrisChinchilla** [comment]
@chriseth Hmm, don't know what happened there, fixed now.

**Review by chriseth — APPROVED** [review]

**Pull request merged**